core/include/gnuradio-4.0/ClaimStrategy.hpp

```diff
@@ -24,14 +24,13 @@ struct NoCapacityException : public std::runtime_error {
 
 template<typename T>
 concept ClaimStrategy = requires(T /*const*/ t, const std::vector<std::shared_ptr<Sequence>> &dependents, const std::size_t requiredCapacity,
-        const std::make_signed_t<std::size_t> cursorValue, const std::make_signed_t<std::size_t> sequence, const std::make_signed_t<std::size_t> availableSequence, const std::size_t n_slots_to_claim) {
+        const std::make_signed_t<std::size_t> cursorValue, const std::make_signed_t<std::size_t> sequence, const std::make_signed_t<std::size_t> availableSequence, const std::size_t n_slots_to_claim, const std::size_t n_slots) {
     { t.hasAvailableCapacity(dependents, requiredCapacity, cursorValue) } -> std::same_as<bool>;
     { t.next(dependents, n_slots_to_claim) } -> std::same_as<std::make_signed_t<std::size_t>>;
     { t.tryNext(dependents, n_slots_to_claim) } -> std::same_as<std::make_signed_t<std::size_t>>;
     { t.getRemainingCapacity(dependents) } -> std::same_as<std::make_signed_t<std::size_t>>;
-    { t.publish(sequence) } -> std::same_as<void>;
-    { t.isAvailable(sequence) } -> std::same_as<bool>;
-    { t.getHighestPublishedSequence(sequence, availableSequence) } -> std::same_as<std::make_signed_t<std::size_t>>;
+    { t.publish(sequence, n_slots) } -> std::same_as<void>;
+    { t.readCursorPosition() } -> std::same_as<std::make_signed_t<std::size_t>>;
```

Comment:
Trying to find out whether it is possible to fix this without chaning the Circular Buffer API. I made an separate Branch with the unit tests and a yet incomplete fix on the Reader side here: https://github.com/fair-acc/graph-prototype/tree/fixCircularBufferNoApiChange

Unfortunately this does not seem to be enough to pass the new multiProducerMap unit tests.